### PR TITLE
Draw boundary dataset and boundary filtering

### PR DESCRIPTION
**Pull request by matthewma7**
Utilize GeoJS annotation layer for boundary dataset creating. 
Uses bounding box support based on the previous PRs for intersecting filtering.

### CODE REVIEW

**Comment by manthey** [comment]
When you delete a boundary that was used to filter, should the filter be turned off?

**Comment by manthey** [comment]
When you are in draw-boundary mode, can you change the draw icon so that it shows it is selected?

File: web_external/views/map/MapPanel.js [hunk 1]
```diff
@@ -318,11 +319,34 @@ const MapPanel = Panel.extend({
...
 
+        this.listenTo(events, 'm:draw-boundary-dataset', () => {
+            if (!this.drawDatasetLayer) {
+                var layer = this.map.createLayer('annotation', {
+                    annotations: ['rectangle'],
+                    showLabels: false
+                });
+                layer.geoOn(geo.event.annotation.state, (e) => {
+                    bootbox.prompt({
+                        title: 'Give dataset a name?',
```
Comment (manthey) [inline web_external/views/map/MapPanel.js]: I would change this title to `Boundary dataset name` (or at least remove the question mark).

Comment (matthewma7) [reply web_external/views/map/MapPanel.js]: Make sense.

**Comment by matthewma7** [comment]
@manthey Thank you for your review. I think it's ok not to remove the boundary when the dataset is removed? 
I think the icon could have changed to indicate it's drawing mode, but I did some new feature related to icons. So, I think I will add it in a following PR. 

When testing, I found out the boundary calculation is not supported on WMS layers, so I added a series of check and message to handle the situation when boundary can't be acquired.

**Comment by manthey** [comment]
If it wouldn't be too much work, can you add a test with a dataset where you can't get bounds?  Otherwise, this looks good.

**Comment by matthewma7** [comment]
Thank you @manthey for reviewing 🍺

**Pull request merged**